### PR TITLE
Support task list priorities in LSP

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Followup to https://github.com/dotnet/roslyn/pull/66293.

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
Going to wait for approval from at least @AdamYoblick and @kayle to make sure client side is ok with this.

File: src/Features/LanguageServer/ProtocolUnitTests/Diagnostics/PullDiagnosticTests.cs
Comment (kayle) [inline src/Features/LanguageServer/ProtocolUnitTests/Diagnostics/PullDiagnosticTests.cs:885]: ImmutableArray.Create($"TODO:{priString}")); should be sufficient as well

**Review by kayle — APPROVED** [review]

**Pull request merged**
